src/NuGet.Services.Metadata.Catalog.Monitoring/Utility/ComparisonUtils.cs

```diff
@@ -0,0 +1,246 @@
+                {
+                    inconsistentProperties.Add(enumerableProperty);
+                }
+            }
+
+            return inconsistentProperties.Any() ? inconsistentProperties : null;
+        }
+
+        /// <summary>
+        /// We only verify these types. This prevents errors with unexpected fields.
+        /// </summary>
+        private static IDictionary<Type, IEqualityComparer<object>> _comparatorsForTypes = new Dictionary<Type, IEqualityComparer<object>>
+        {
+            { typeof(bool), new BoolComparer() },
+            { typeof(string), new StringComparer() },
+            { typeof(DateTimeOffset?), new DateTimeOffsetComparer() },
+            { typeof(NuGetVersion), new CastingComparer<NuGetVersion>() },
+            { typeof(Uri), new CastingComparer<Uri>() }
+        };
+
+        #region Comparers
+        /// <summary>
+        /// Returns true if both bools are equal.
+        /// Returns false otherwise.
+        /// </summary>
+        private class BoolComparer : IEqualityComparer<object>
+        {
+            public new bool Equals(object x, object y)
+            {
+                var boolX = (bool)x;
+                var boolY = (bool)y;
+
+                return boolX == boolY;
+            }
+
+            public int GetHashCode(object obj)
+            {
+                return obj.GetHashCode();
+            }
+        }
+
+        /// <summary>
+        /// Returns true
+        ///   if both <see cref="string"/>s are null or empty,
+        ///   if both <see cref="string"/>s can be parsed into <see cref="Uri"/>s and their <see cref="Uri.ToString"/>s return equal,
+        ///   or if <see cref="string.Equals(string)"/> returns true.
+        ///   
+        /// Returns false otherwise.
+        /// 
+        /// Note that we consider null and empty equal because for fields without a value,
+        /// the V2 implementation returns empty string and the V3 implementation returns null.
+        /// </summary>
+        private class StringComparer : IEqualityComparer<object>
+        {
+            public new bool Equals(object x, object y)
+            {
+                var stringX = (string)x;
+                var stringY = (string)y;
+
+                // Consider null and empty equal.
+                // For fields without a value, the V2 implementation returns empty string and the V3 implementation returns null.
+                if (string.IsNullOrEmpty(stringX))
+                {
+                    return string.IsNullOrEmpty(stringY);
+                }
+
+                /// Try to parse the strings as <see cref="Uri"/>s and then compare them on that.
+                /// The V2 and V3 implementations sometimes returns <see cref="Uri"/>s in different forms (i.e. trailing slash).
+                if (Uri.TryCreate(stringX, UriKind.Absolute, out var uriX) && Uri.TryCreate(stringY, UriKind.Absolute, out var uriY))
```

Comment:
Is there another way to distinguish Url properties, instead of attempting Uri.TryCreate on all string props? Property naming (ends with "Url"), or whitelist?